solutions/USACO open20-haircut.mdx

```diff
@@ -0,0 +1,167 @@
+---
+id: usaco-1041
+source: USACO Gold 2020 US Open
+title: Haircut
+author: Jeffrey Meng
+---
+
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_haircut_gold_open20.html)
+
+First, we need to count the number of inversions. (Fun fact: the number of inversions is also a measure of how sorted an array is. If an array has 0 inversions, it's perfectly sorted in ascending order.) 
+
+One way to count inversions is to use a Binary Indexed Tree. Think of an inversion as a pair $i, j$, with $i < j$ & $A[i] > A[j]$. Then, we can easily compute, for each element in the array, the number of inversions it is $j$ for.
+
+To do this, we use a BIT as a frequency table, keeping track of the number of hairs before index $i$ of each possible value. If our BIT is named `tree`, `tree[i]` will tell us how many times $i$ is seen in the array of hairs. As long as we fill out the frequency table in the order the elements appear in the array, we can use the frequency table to also count the number of inversions, because the frequency table only contains hairs to the left of the hair we are currently considering.
+
+That leads us to this algorithm:
+
```

Comment:
Maybe get rid of these newlines between lists? Idk whether it changes the spacing on the webpage, but I think it'll make it look nicer in markdown form (it's up to you to decide what you prefer though)

Reply:
@thecodingwizard the nested list doesn't work without the spaces for some reason

Reply:
Use 1. Rather than a.